src/com/mendix/widget/carousel/components/__tests__/Carousel.spec.tsx

```diff
@@ -0,0 +1,49 @@
+import { shallow } from "enzyme";
+import * as React from "react";
+
+import { Carousel, CarouselProps } from "../Carousel";
+
+describe("Carousel component", () => {
+    const staticImages = [
+        { imageUrl: "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" },
+        { imageUrl: "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" }
+    ];
+    const render = (props: CarouselProps) => shallow(<Carousel {...props} />);
+
+    it("should render carousel structure", () => {
+        const output = render({ images: staticImages });
+        expect(output).toMatchStructure(
+            <div className="carousel">
+                <div className="carousel-inner" />
+            </div>
+        );
+    });
+
+    it("should render carousel with 2 children", () => {
+        const wrapper = render({ images: staticImages });
+        expect(wrapper.find(".carousel-inner").children().length).toEqual(staticImages.length);
+    });
+
+    it("should render carousel with the class 'carousel'", () => {
+        const output = render({ images: staticImages });
+        expect(output).toHaveClass("carousel");
+    });
+
+    it("should render carousel child to have class 'carousel-inner'", () => {
+        const output = render({ images: staticImages });
+        expect(output.children()).toHaveClass("carousel-inner");
+    });
+
+    it("first child should be active", () => {
+        const wrapper = render({ images: staticImages });
+        const children = wrapper.find(".carousel-inner").children();
+        expect(children.first().prop("active")).toBe(true);
+    });
+
+    it("should have only one active child", () => {
+        const wrapper = render({ images: staticImages });
+        const children = wrapper.find(".carousel-inner").children();
+        const activeChildren = children.filterWhere(c => c.prop("active") === true);
```

Comment:
I'm maintaining `filterWhere()` which takes in a predicate function. The `filter()` function only accepts a selector, similar to `find()`. Tests failed on all variations I tried that use `filter()`, including the one above.

Moved `children` inline.